lib/axiom/fs/stream/post_message_streams.js

```diff
@@ -0,0 +1,183 @@
+  } else {
+    this.connectionState_ = GenericStreams.ConnectionState.CONNECTING;
+    port.onMessage.addListener(this.onConnectedMessage_, false);
+    port.onDisconnect.addListener(this.onConnectionError_, false);
+
+    return new Promise(function (resolve, reject) {
+      this.onOpenResolve_ = resolve;
+      this.onOpenReject_ = reject;
+    }.bind(this));
+  }
+
+};
+
+/**
+ * @return {void}
+ */
+PostMessageStreams.prototype.close = function() {
+  this.port.disconnect();
+  this.connectionState_ = GenericStreams.ConnectionState.CLOSING;
+  //TODO: Implement
+};
+
+/**
+ * @protected
+ * @param {EventWithCallback|string} item
+ * @return {void}
+ */
+PostMessageStreams.prototype.handleSend_ = function(item) {
+  GenericStreams.prototype.handleSend_.call(this, item);
+
+  if (item instanceof EventWithCallback) {
+    this.port.postMessage(/* string */(item.value), undefined);
+
+    // TODO (ericarnold): implement callback
+    item.callback();
+  } else {
+    this.port.postMessage(/** @type {string} */(item), undefined);
+  }
+};
+
+/**
+ * @return {void}
+ */
+PostMessageStreams.prototype.onConnectedMessage_ = function(event) {
+  this.cleanupConnectionListeners_();
+  this.handleConnected_();
+  this.onOpenResolve_();
+}
+
+/**
+ * @return {void}
+ */
+PostMessageStreams.prototype.onConnectionError_ = function(event) {
+  this.connectionState_ = GenericStreams.ConnectionState.CLOSED;
+  this.cleanupConnectionListeners_();
+  this.onOpenReject_(new AxiomError.Runtime('Error connecting to extension'));
+  this.port = null;
+}
+
+/**
+ * @return {void}
+ */
+PostMessageStreams.prototype.onDisconnect_ = function(event) {
+  this.connectionState_ = GenericStreams.ConnectionState.CLOSED;
+  this.handleClose_();
+  this.port = null;
+}
+
+/**
+ * @protected
+ * @override
+ * @return {void}
+ */
+PostMessageStreams.prototype.handleConnected_ = function() {
+  this.connectionState_ = GenericStreams.ConnectionState.CONNECTED;
+  this.port.onMessage.addListener(this.onPostMessage_, false);
+  this.port.onDisconnect.addListener(this.onDisconnect_, false);
+  GenericStreams.prototype.handleConnected_.call(this);
+}
+
+/**
+ * @protected
+ * @return {void}
+ */
+PostMessageStreams.prototype.handleClose_ = function() {
+  this.port.onMessage.removeListener(this.onPostMessage_, false);
+  this.port.onDisconnect.removeListener(this.onDisconnect_, false);
+  GenericStreams.prototype.handleClose_.call(this);
+}
+
+/**
+ * @return {void}
+ */
+PostMessageStreams.prototype.cleanupConnectionListeners_ = function() {
+  this.port.onMessage.removeListener(this.onConnectedMessage_, false);
+  this.port.onDisconnect.removeListener(this.onConnectionError_, false);
+}
+
+/**
+ * @return {void}
+ */
+PostMessageStreams.prototype.onPostMessage_ = function(message) {
+  this.handleReceive_(message);
+};
+
+/**
+ * Return the state of the connection.
+ *
+ * @protected
+ */
+PostMessageStreams.prototype.getState_ = function() {
+  if (this.port == null) {
```

Comment:
`this.port === null`. With `==`, `undefined == null` evaluates to `true` (just one example).